### PR TITLE
Included some C# Keywords

### PR DESCRIPTION
**Pull request by lucaslra**
- Included some Clojure Extensions
- Included one Powershell Extension
- Included some Csharp Keywords
- Reorganization of the language.main.js define call (To match the folders order)

### CODE REVIEW

File: extensions/clojure/syntaxes/Clojure.tmLanguage [hunk 1]
```diff
@@ -7,6 +7,9 @@
 		<string>clj</string>
 		<string>cljs</string>
 		<string>clojure</string>
+		<string>cljc</string>
+		<string>cljx</string>
+		<string>edn</string>
```
Comment (aeschli) [inline extensions/clojure/syntaxes/Clojure.tmLanguage]: If you want to add file extensions for clojure, you have to do that `/extensions/clojure/package.json`. From `/extensions/clojure/syntaxes/Clojure.tmLanguage` we only take the colorizer information.

File: extensions/powershell/syntaxes/PowershellSyntax.tmLanguage [hunk 2]
```diff
@@ -7,6 +7,7 @@
 		<string>ps1</string>
 		<string>psm1</string>
 		<string>psd1</string>
+		<string>ps1xml</string>
```
Comment (aeschli) [inline extensions/powershell/syntaxes/PowershellSyntax.tmLanguage]: Same here, `/extensions/powershell/package.json` contains the file extensions

File: src/vs/base/browser/browser.ts [hunk 3]
```diff
@@ -48,6 +48,7 @@ var globals = <IGlobalScope><any> (typeof self === 'object' ? self : global);
 // IE: "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1; WOW64; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E; MS-RTC LM 8; InfoPath.3; Zune 4.7)"
 // Opera:	"Opera/9.80 (Windows NT 6.1; U; en) Presto/2.9.168 Version/11.52"
 // FF: "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:8.0) Gecko/20100101 Firefox/8.0"
+// Edge: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246"
```
Comment (aeschli) [inline src/vs/base/browser/browser.ts]: I don't think we need that comment at the moment

File: src/vs/languages/languages.main.js [hunk 4]
```diff
@@ -8,13 +8,13 @@ define([
...
+	"vs/languages/less/common/less.contribution",
 	"vs/languages/markdown/common/markdown.contribution",
 	"vs/languages/php/common/php.contribution",
 	"vs/languages/plaintext/common/plaintext.contribution",
 	"vs/languages/razor/common/razor.contribution",
-	"vs/languages/typescript/common/typescript.contribution",
-	"vs/languages/less/common/less.contribution",
-	"vs/languages/sass/common/sass.contribution"
+	"vs/languages/sass/common/sass.contribution",
+	"vs/languages/typescript/common/typescript.contribution"
```
Comment (aeschli) [inline src/vs/languages/languages.main.js]: This change is not really necessary

File: src/vs/editor/standalone-languages/csharp.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ export var language = <ILanguage> {
 		'internal', 'private', 'abstract', 'sealed', 'static', 'struct', 'readonly',
 		'volatile', 'virtual', 'override', 'params', 'get', 'set', 'add', 'remove',
 		'operator', 'true', 'false', 'implicit', 'explicit', 'interface', 'enum',
-		'null', 'async', 'await'
+		'null', 'async', 'await','fixed','sizeof','stackalloc','unsafe'
 		],
```
Comment (aeschli) [inline src/vs/editor/standalone-languages/csharp.ts]: This change is good, but note that this file is not in use in VS Code. For CSharp colorization in VS Code we use `/extensions/csharp/syntaxes/csharp.json`. 
This file is used for the standalone editor build.

**Comment by aeschli** [comment]
@lucaslra Thanks for the contribution. Only the change to `src/vs/editor/standalone-languages/csharp.ts` is valid. If you remove the other changes I can accept and merge the pull request.

**Comment by lucaslra** [comment]
Hi @aeschli, thanks for the feedbacks. Is there a document where the architecture of the VS Code source is described? That should help me to avoid these type of pull requests.

**Comment by aeschli** [comment]
@lucaslra Sorry, except for the extension API extension we currently have no further documentation on API or architecture. But it's a good point, we should definitly start a high level overview in the wiki.

**Comment by lucaslra** [comment]
I think it will greatly improve the quality of the contributions.
I made the changes you requested, please let me know if its OK now.

**Comment by aeschli** [comment]
@lucaslra Please make only minimal changes. In this case, only have a change in `src/vs/editor/standalone-languages/csharp.ts`.

**Comment by aeschli** [comment]
@lucaslra Thanks a lot, sorry for the back and forth.
Good point on providing more guidelines and documentation. I started by updating https://github.com/Microsoft/vscode/wiki/How-to-Contribute

**Pull request merged**